tests/template_test.go

```diff
@@ -0,0 +1,91 @@
+package tests_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"kubevirt.io/kubevirt-ansible/tests"
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+const (
+	fedoraCloudImageDev  string = "registry:5000/kubevirt/fedora-cloud-registry-disk-demo:devel"
+	fedoraCloudImageTest string = "kubevirt/fedora-cloud-registry-disk-demo:latest"
+)
+
+var _ = Describe("Virtual machine template tests", func() {
+	ns := ktests.NamespaceTestDefault
+
+	var vm tests.VMManifest
+
+	BeforeEach(func() {
+		ktests.BeforeTestCleanup()
+	})
+
+	AfterEach(func() {
+		err := os.RemoveAll(filepath.Dir(vm.Manifest))
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Describe("Test fedora template", func() {
+		BeforeEach(func() {
+			dir, err := ioutil.TempDir("", "vm-template-")
+			Expect(err).ToNot(HaveOccurred())
+
+			vm.Manifest = filepath.Join(dir, "vm-template-fedora.yaml")
+			vm.Name = "fedora"
+			vm.Template = "vm-template-fedora"
+			vm.Params = []string{"-p", "NAME=" + vm.Name, "-p", "CPU_CORES=2", "-p", "MEMORY=2048Mi"}
+
+			if _, err := vm.ProcessClusterTemplate(); err != nil {
+				Skip(fmt.Sprintf("error occurrs when process template %s in namespace %s", vm.Template, tests.TemplateNS))
```

Comment:
It looks `Skipf` is not defined in ginkgo when I try to use it.

Reply:
Although Ginkgo defines `Skipf` in the interface, it indeed does not implement the function yet. I was mistaken because `Skipf` is often used by the Kubernetes test framework. It turned out that Kubernetes defines it as an [util function](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go):

```go
func Skipf(format string, args ...interface{}) { 
    msg := fmt.Sprintf(format, args...)
    log("INFO", msg)
    ginkgowrapper.Skip(nowStamp() + ": " + msg)
} 
```